src/libraries/Common/src/System/Security/Cryptography/CompositeMLDsaManaged.cs

```diff
@@ -0,0 +1,682 @@
+        {
+            // draft-ietf-lamps-pq-composite-sigs-latest (June 20, 2025), 4.2
+            //  1.  If len(ctx) > 255:
+            //      return error
+
+            Debug.Assert(context.Length <= 255, $"Caller should have checked context.Length, got {context.Length}");
+
+            //  2.  Compute the Message representative M'.
+            //      As in FIPS 204, len(ctx) is encoded as a single unsigned byte.
+            //      Randomize the message representative
+            //
+            //          r = Random(32)
+            //          M' :=  Prefix || Domain || len(ctx) || ctx || r
+            //                                              || PH( M )
+
+            Span<byte> r = stackalloc byte[CompositeMLDsaAlgorithm.RandomizerSizeInBytes];
+            RandomNumberGenerator.Fill(r);
+
+            byte[] M_prime = GetMessageRepresentative(AlgorithmDetails, context, r, data);
+
+            //  3.  Separate the private key into component keys
+            //      and re-generate the ML-DSA key from seed.
+            //
+            //          (mldsaSeed, tradSK) = DeserializePrivateKey(sk)
+            //          (_, mldsaSK) = ML-DSA.KeyGen(mldsaSeed)
+
+            /* no-op */
+
+            //  4.  Generate the two component signatures independently by calculating
+            //      the signature over M' according to their algorithm specifications.
+            //
+            //          mldsaSig = ML-DSA.Sign( mldsaSK, M', ctx=Domain )
+            //          tradSig = Trad.Sign( tradSK, M' )
+
+            //  Note that in step 4 above, both component signature processes are
+            //  invoked, and no indication is given about which one failed.This
+            //  SHOULD be done in a timing-invariant way to prevent side-channel
+            //  attackers from learning which component algorithm failed.
+
+            Span<byte> randomizer = destination.Slice(0, CompositeMLDsaAlgorithm.RandomizerSizeInBytes);
+            Span<byte> mldsaSig = destination.Slice(CompositeMLDsaAlgorithm.RandomizerSizeInBytes, AlgorithmDetails.MLDsaAlgorithm.SignatureSizeInBytes);
+            Span<byte> tradSig = destination.Slice(CompositeMLDsaAlgorithm.RandomizerSizeInBytes + AlgorithmDetails.MLDsaAlgorithm.SignatureSizeInBytes);
+
+            bool mldsaSigned = false;
+            bool tradSigned = false;
+
+            try
+            {
+                _mldsa.SignData(M_prime, mldsaSig, AlgorithmDetails.DomainSeparator);
+                mldsaSigned = true;
+            }
+            catch (CryptographicException)
+            {
+            }
+
+            int tradBytesWritten = 0;
+
+            try
+            {
+                tradSigned = _componentAlgorithm.TrySignData(M_prime, tradSig, out tradBytesWritten);
+                Debug.Assert(tradSigned, "Signature buffer should be exactly sized.");
```

Comment:
I don't understand this assert.  My recollection is that the base class currently doesn't do a defensive up-rent to MaximumSignatureSize (though I did suggest that it should).  Is this assert "I've only written RSA"?

Reply:
Yeah - assert is right, but message is wrong. It should just say the caller should have made sure buffer is long enough.